### PR TITLE
feat: checkin-checkout

### PR DESCRIPTION
**Pull request by dan-lv**
I build check-in, check-out feature follow repository pattern. 1 user check-in, check-out one time per day. user must check-in before checking-out.

### CODE REVIEW

File: app/Contracts/Interfaces/CheckInInterface.php [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?php
+namespace App\Contracts\Interfaces;
+
+interface CheckInInterface
+{
+	public function checkIn();
```
Comment (cuongzinza) [inline app/Contracts/Interfaces/CheckInInterface.php]: indent 4 spaces

File: app/Contracts/Interfaces/CheckOutInterface.php [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?php
+namespace App\Contracts\Interfaces;
+
+interface CheckOutInterface
+{
+	public function checkOut();
```
Comment (cuongzinza) [inline app/Contracts/Interfaces/CheckOutInterface.php]: indent 4 spaces

File: app/Contracts/Repositories/CheckInRepository.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<?php
+namespace App\Contracts\Repositories;
+
+use App\Contracts\Interfaces\CheckInInterface;
+use App\Models\User;
+use App\Models\TimeLog;
+use Carbon\Carbon;
+use Auth;
+
+class CheckInRepository implements CheckInInterface
```
Comment (cuongzinza) [inline app/Contracts/Repositories/CheckInRepository.php]: repository is for doing query, not logic or redirection

File: app/Http/Controllers/User/CheckInController.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+     * @return \Illuminate\Http\Response
+     */
+    private $checkInRepository;
+
+    public function __construct(CheckInInterface $checkInRepository)
+    {
+        $this->checkInRepository = $checkInRepository;
+    }
+
+    public function index()
```
Comment (cuongzinza) [inline app/Http/Controllers/User/CheckInController.php]: if functions are not used, remove

File: app/Http/Controllers/User/CheckInController.php [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        //
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
```
Comment (cuongzinza) [inline app/Http/Controllers/User/CheckInController.php]: if functions are not used, remove

File: app/Http/Controllers/User/CheckInController.php [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        return $this->checkInRepository->checkIn();
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
```
Comment (cuongzinza) [inline app/Http/Controllers/User/CheckInController.php]: if functions are not used, remove

File: app/Http/Controllers/User/CheckInController.php [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        //
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
```
Comment (cuongzinza) [inline app/Http/Controllers/User/CheckInController.php]: if functions are not used, remove

File: app/Http/Controllers/User/CheckInController.php [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
```
Comment (cuongzinza) [inline app/Http/Controllers/User/CheckInController.php]: if functions are not used, remove

File: resources/sass/mystyle.scss [hunk 9]
```diff
@@ -152,4 +152,10 @@ h3.text {
 .btn-position {
 	top: 500px;
```
Comment (cuongzinza) [inline resources/sass/mystyle.scss]: check indent

File: resources/sass/mystyle.scss [hunk 10]
```diff
@@ -152,4 +152,10 @@ h3.text {
...
     position: relative;
-}
\ No newline at end of file
+}
+.status {
+	text-align: center;
```
Comment (cuongzinza) [inline resources/sass/mystyle.scss]: check indent

File: resources/views/user/home.blade.php [hunk 11]
```diff
@@ -1,15 +1,23 @@
...
 <div class="image">
+	@if (session('status'))
+	<p class="status">{{ session('status') }}</p>
+	@endif
 	<h2 class="text">Find Creativity In</h2>
 	<h3 class="text">Gray World</h3>
 	<div class="container btn-position">
 		<div class="wrap">
-			<a href="" class="item1"> Check-in </a>
+			<a href="" class="item1" onclick="event.preventDefault(); document.getElementById('check-in').submit();"> Check-in </a>
```
Comment (cuongzinza) [inline resources/views/user/home.blade.php]: why not using form here??

File: app/Contracts/Repositories/CheckInRepository.php [hunk 12]
```diff
@@ -0,0 +1,31 @@
+<?php
+namespace App\Contracts\Repositories;
```
Comment (cuongzinza) [inline app/Contracts/Repositories/CheckInRepository.php]: `Contracts` and `Repositories` is different, they should be on the same level

File: resources/views/user/home.blade.php (hunk 11 above)
Comment (dan-lv) [reply resources/views/user/home.blade.php]: i will use for popup-confirm after that

File: resources/views/user/home.blade.php [hunk 13]
```diff
@@ -1,15 +1,23 @@
 @extends('user.layout')
```
Comment (cuongzinza) [inline resources/views/user/home.blade.php]: blade files are using tab indentation, but php files are using spaces
please unify the style

File: routes/web.php [hunk 14]
```diff
@@ -18,3 +18,13 @@
 Auth::routes();
```
Comment (cuongzinza) [inline routes/web.php]: this too is using tab indentation

File: routes/web.php [hunk 15]
```diff
@@ -18,3 +18,13 @@
...
 
 Route::get('/home', 'HomeController@index')->name('home');
+
+Route::group([
+	'namespace' => 'User',
+	'middleware' => 'auth',
+], function() {
+	Route::resources([
+		'check-ins' => 'CheckInController',
```
Comment (cuongzinza) [inline routes/web.php]: `->only(['store'])` ?

File: routes/web.php [hunk 16]
```diff
@@ -18,3 +18,13 @@
...
+		'check-outs' => 'CheckOutController',
```
Comment (cuongzinza) [inline routes/web.php]: `->only(['store'])` ?

File: app/Http/Controllers/User/CheckInController.php [hunk 17]
```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace App\Http\Controllers\User;
+
+use App\Http\Controllers\Controller;
+use App\Interfaces\TimeLogInterface;
+
+class CheckInController extends Controller
+{
+    private $TimeLogRepository;
```
Comment (cuongzinza) [inline app/Http/Controllers/User/CheckInController.php]: first letter is lowercase!

File: app/Http/Controllers/User/CheckInController.php [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+
+    public function __construct(TimeLogInterface $TimeLogRepository)
+    {
+        $this->TimeLogRepository = $TimeLogRepository;
+    }
+
+    public function store()
+    {
+        $check_time_log = $this->TimeLogRepository->getTimeLogToDay();
```
Comment (cuongzinza) [inline app/Http/Controllers/User/CheckInController.php]: variable name is camelCase

`ToDay` --> `Today`??

File: app/Http/Controllers/User/CheckInController.php [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+
+        if (!$check_time_log) {
+            $this->TimeLogRepository->setCheckIn();
+
+            return redirect('/')->with('status', trans('time_log.check_in_success'));
```
Comment (cuongzinza) [inline app/Http/Controllers/User/CheckInController.php]: can use function `__()`

File: app/Http/Controllers/User/CheckOutController.php [hunk 20]
```diff
@@ -0,0 +1,38 @@
+<?php
```
Comment (cuongzinza) [inline app/Http/Controllers/User/CheckOutController.php]: same for the comments in `CheckInController`

File: app/Repositories/TimeLogRepository.php [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+
+use App\Interfaces\TimeLogInterface;
+use App\Models\User;
+use App\Models\TimeLog;
+use Carbon\Carbon;
+use Auth;
+
+class TimeLogRepository implements TimeLogInterface 
+{
+    public function setTime()
```
Comment (cuongzinza) [inline app/Repositories/TimeLogRepository.php]: this should be `getTime()`
and maybe this function is ok to be private

File: app/Repositories/TimeLogRepository.php [hunk 22]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        return Carbon::now();
+    }
+
+    public function getUser()
```
Comment (cuongzinza) [inline app/Repositories/TimeLogRepository.php]: this too can be private

File: app/Repositories/TimeLogRepository.php [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        return Auth::user();
+    }
+
+    public function createTimeLog()
```
Comment (cuongzinza) [inline app/Repositories/TimeLogRepository.php]: this too can be private

**Comment by cuongzinza** [comment]
@dan-lv and please use bootstrap

File: app/Http/Controllers/User/CheckInController.php (hunk 19 above)
Comment (dan-lv) [reply app/Http/Controllers/User/CheckInController.php]: well, good information for me

File: app/Interfaces/TimeLogInterface.php [hunk 24]
```diff
@@ -0,0 +1,9 @@
+<?php
+namespace App\Interfaces;
+
+interface TimeLogInterface
+{
+    public function getTimeLogToday();
+    public function setCheckIn();
+    public function setCheckOut(object $check_time_log);
```
Comment (cuongzinza) [inline app/Interfaces/TimeLogInterface.php]: variable name should be camelCased

File: app/Repositories/TimeLogRepository.php [hunk 25]
```diff
@@ -0,0 +1,51 @@
...
+    public function setCheckIn() { 
+        
+        $this->createTimeLog()->create([
+            'check_in' => $this->getTime()->toTimeString(),
+            'day' => $this->getTime()->toDateString(),
+            'user_id' => $this->getUser()->id,
+        ]);
+    }
+
+    public function setCheckOut($check_time_log) {
```
Comment (cuongzinza) [inline app/Repositories/TimeLogRepository.php]: variable name should be camelCased

**Pull request merged**